### PR TITLE
Parse HTTP Link headers for multiple links

### PR DESCRIPTION
**Pull request by WhiteCat22**
Parse HTTP Link headers for multiple links on Jaxrs Response objects and create a FAT test.

https://github.com/OpenLiberty/open-liberty/issues/8863

### CODE REVIEW

**Comment by WhiteCat22** [comment]
#build

File: dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/linkheader/src/com/ibm/ws/jaxrs/fat/linkheader/ClientTestServlet.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        System.out.println("Links:");
+        for (Link link : links) {
+            System.out.println(link);
+        }
+
+        Link link = response.getLink("next");
+        System.out.println("next:");
+        System.out.println(link);
+    }
```
[the remark this answers is not in the document]
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/linkheader/src/com/ibm/ws/jaxrs/fat/linkheader/ClientTestServlet.java]: Ah, you are right, I should do something explicit with the links. Previously, there would be an exception thrown here and the test would blow up prior to the fix.

File: dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/impl/ResponseImpl.java [hunk 2]
```diff
@@ -323,10 +326,19 @@ public boolean hasLink(String relation) {
...
         if (linkValues != null) {
             for (Object o : linkValues) {
-                Link link = o instanceof Link ? (Link) o : Link.valueOf(o.toString());
-                if (relation.equals(link.getRel())) {
+                // Liberty Change Start
+                if (o instanceof Link && relation.equals(((Link)o).getRel())) {
                     return true;
                 }
+                
+                String[] links = o.toString().split(",(?=\\<|$)");
```
[the remark this answers is not in the document]
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/impl/ResponseImpl.java]: good catch, I updated to use `linkDelimeter` instead.

**Comment by WhiteCat22** [comment]
#build

**Comment by bmhm** [comment]
Sorry about the test case I missed and the regex pattern not being a constant. Thanks for reviewing, fixing and merging! :)

**Comment by WhiteCat22** [comment]
Thank you for bringing to light the issue and contributing the solution @bmhm

**Comment by bmhm** [comment]
@WhiteCat22 I created a pull request based on your changes for Apache CXF: https://github.com/apache/cxf/pull/610

Please notice that those changes may not be compatible, as you violated CXF's checkstyle by moving the `static final` variable below the instance variables. Also the lower-camel-case name pattern is not allowed for static final variables in CXF.

I also had to rework an `ArrayList` to just use `List` for a variable.

**Pull request merged**